src/aggregator/aggregator/aggregator.go

```diff
@@ -799,45 +809,78 @@ func (m *aggregatorAddMetricMetrics) ReportError(err error) {
 	}
 }
 
-type aggregatorAddUntimedMetrics struct {
-	aggregatorAddMetricMetrics
+type aggregatorAddUntimedErrorMetrics struct {
+	aggregatorAddMetricErrorMetrics
 
 	invalidMetricTypes tally.Counter
 }
 
+func newAggregatorAddUntimedErrorMetrics(
+	scope tally.Scope,
+) aggregatorAddUntimedErrorMetrics {
+	return aggregatorAddUntimedErrorMetrics{
+		aggregatorAddMetricErrorMetrics: newAggregatorAddMetricErrorMetrics(scope),
+		invalidMetricTypes: scope.Tagged(map[string]string{
+			"reason": "invalid-metric-types",
+		}).Counter("errors"),
+	}
+}
+
+type aggregatorAddUntimedMetrics struct {
+	aggregatorAddMetricSuccessMetrics
+
+	leaderErrors    aggregatorAddUntimedErrorMetrics
+	nonLeaderErrors aggregatorAddUntimedErrorMetrics
+}
+
 func newAggregatorAddUntimedMetrics(
 	scope tally.Scope,
 	opts instrument.TimerOptions,
 ) aggregatorAddUntimedMetrics {
+	var (
+		leaderScope = scope.Tagged(map[string]string{
+			"role": "leader",
+		})
+		nonLeaderScope = scope.Tagged(map[string]string{
+			"role": "non-leader",
+		})
+	)
 	return aggregatorAddUntimedMetrics{
-		aggregatorAddMetricMetrics: newAggregatorAddMetricMetrics(scope, opts),
-		invalidMetricTypes: scope.Tagged(map[string]string{
-			"reason": "invalid-metric-types",
-		}).Counter("errors"),
+		aggregatorAddMetricSuccessMetrics: newAggregatorAddMetricSuccessMetrics(scope, opts),
+
+		leaderErrors:    newAggregatorAddUntimedErrorMetrics(leaderScope),
+		nonLeaderErrors: newAggregatorAddUntimedErrorMetrics(nonLeaderScope),
 	}
 }
 
-func (m *aggregatorAddUntimedMetrics) ReportError(err error) {
+func (m *aggregatorAddUntimedMetrics) ReportError(err error, role ElectionState) {
+	var errors *aggregatorAddUntimedErrorMetrics
+	switch role {
+	case LeaderState:
+		errors = &m.leaderErrors
+	default:
+		errors = &m.nonLeaderErrors
+	}
```

Comment:
Nit - consider simplifying to (here and in other places):
```suggestion
	errors := &m.nonLeaderErrors
	if role == LeaderState {
		errors = &m.leaderErrors
	}
```

Reply:
https://github.com/m3db/m3/pull/3559/commits/f7dac96dc1a995ee2d63a12ab7e657a51928138e